File: Data Structures & Algorithms/majority-element/submission-0.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) 
    {
        int major=INT_MIN,result=0;
        unordered_map<int,int> mp;
        for(auto &num : nums)
        {
            mp[num]++;
        }        
        for(auto &ele:mp)
        {
            if(major<ele.second)
            {
                result=ele.first;
                major=ele.second;
            }
        }
        return result;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) 
    {
        int p=1;
        int arr[]=new int[nums.length];
        for(int i=0;i<nums.length;i++)
        {
            p=1;
            for(int j=0;j<nums.length;j++)
            {
                if(i==j)
                {
                    continue;
                }
                p=p*nums[j];
            }
            arr[i]=p;
        }       
        return arr; 
    }
}  
